### PR TITLE
fix(cli): fix type reference schemas

### PR DESCRIPTION
**Pull request by dsinghvi**
## Human Summary
- Fixes a bug where we updated the OpenAPI converter to generate invalid type references with titles
- Refactor `TypeReferenceSchema.ts` so that its easier to add to over time

## AI Generated Summary

This PR introduces changes to the `TypeReferenceSchema` and `VariableDeclarationSchema` in the `fern-definition` and `openapi-ir-to-fern` packages.

## Changes
- The `TypeReferenceSchema` is now imported from the `TypeReferenceSchema.ts` file instead of the `TypeReferenceWithDocsAndAvailabilitySchema.ts` file.
- The `VariableDeclarationSchema` is now defined as a union of a string and a strict object with a `type` property and an extension of the `WithDocsSchema.shape`.
- The `TypeReferenceSchema` is now exported as a constant and a type, with the constant being defined as the result of calling the `extendTypeReferenceSchema` function with an object shape as an argument.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAndDisplayNameAndAvailabilitySchema` is now returned by the `buildPrimitiveTypeReference`, `buildBooleanTypeReference`, `buildLongTypeReference`, `buildStringTypeReference`, `buildIntegerTypeReference`, `buildFloatTypeReference`, `buildDoubleTypeReference`, `buildReferenceTypeReference`, `buildArrayTypeReference`, `buildMapTypeReference`, `buildOptionalTypeReference`, `buildUnknownTypeReference`, `buildLiteralTypeReference`, `buildEnumTypeReference`, `buildObjectTypeReference`, and `buildOneOfTypeReference` functions.
- The `TypeReferenceWithDocsAnd

### CODE REVIEW

File: packages/scripts/package.json [hunk 1]
```diff
@@ -29,32 +29,33 @@
...
-  "dependencies": {
-    "@fern-api/fs-utils": "workspace:*",
-    "@fern-api/fern-definition-json-schema": "workspace:*",
-    "chalk": "^5.3.0",
-    "execa": "^5.1.1",
-    "immer": "^9.0.15",
-    "js-yaml": "^4.1.0",
-    "lodash-es": "^4.17.21",
-    "yargs": "^17.4.1"
+    "dist:cli": "pnpm compile && tsup ./src/cli.ts --format cjs",
```
Comment (dsinghvi) [inline packages/scripts/package.json:32]: moved our script cli to tsup and off of the old stuff

File: packages/scripts/src/cli.ts [hunk 2]
```diff
@@ -18,7 +22,12 @@ void yargs(hideBin(process.argv))
                 demandOption: true
             }),
         async (argv) => {
-            await writeFernJsonSchema(resolve(cwd(), argv.filepath));
+            const filepath = argv.filepath;
+            const jsonSchema = zodToJsonSchema(DefinitionFileSchema, "Fern Definition");
+            const jsonSchemaStr = JSON.stringify(jsonSchema);
+            const config = (await prettier.resolveConfig(filepath)) ?? undefined;
+            const jsonSchemaFormatted = prettier.format(jsonSchemaStr, { ...config, filepath });
+            await writeFile(filepath, jsonSchemaFormatted);
```
Comment (amckinney) [inline packages/scripts/src/cli.ts:30]: nit: I'd consider consolidating this into fewer lines to keep things a little more terse, but I get it if you like the readability as-is.

**Review by amckinney — APPROVED** [review]

File: packages/scripts/src/cli.ts (hunk 2 above)
Comment (dsinghvi) [reply packages/scripts/src/cli.ts:30]: i just kept the function the same as what it used to be (just copy pastad it)

Comment (dsinghvi) [reply packages/scripts/src/cli.ts:30]: will just leave as is to keep things moving

**Pull request merged**